### ISSUE TITLE
The `require-passing-this` rule is also applied to the stack

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#78]
### Describe the bug

The `require-passing-this` rule is supposed to apply only to Construct, but this rule has been applied to stack as well.

### Steps to reproduce

```ts
const app = new cdk.App();
new SampleStack(app, 'Sample')
```

### Expected Behavior

Expected no errors.

### Current Behavior

An error occurred 

### eslint-cdk-plugin Version

1.0.1

### Node.js Version

22.11.0

### Language

TypeScript

### Checklist

- [X] I have searched for similar issues before creating this one
- [X] I have provided all the necessary information to reproduce the issue
- [X] I have verified this issue is related to eslint-cdk-plugin and not another dependency
